omc3/tune_analysis/kick_file_modifiers.py

```diff
@@ -0,0 +1,258 @@
+def get_odr_data(kickac_df, action_plane, tune_plane, order, corrected=False):
+    """ Extract the data from kickac.
+
+    Args:
+        kickac_df: Dataframe containing the data
+        action_plane: Plane of the action
+        tune_plane: Plane of the tune
+        order: Order of the odr fit
+        corrected: (BBQ) corrected data or uncorrected fit?
+
+    Returns:
+        Dictionary containing
+
+    """
+
+    header_val, header_err = _get_odr_headers(corrected)
+    odr_data = DotDict(beta=[0] * (order+1), sd_beta=[0] * (order+1))
+    for idx in range(order+1):
+        odr_data.beta[idx] = kickac_df.headers[header_val(action_plane, tune_plane, idx)]
+        odr_data.sd_beta[idx] = kickac_df.headers[header_err(action_plane, tune_plane, idx)]
+    return odr_data
+
+
+def get_ampdet_data(kickac_df, action_plane, tune_plane, corrected=False):
+    """ Extract the data needed for the (un)corrected amplitude detuning
+    from the kickac dataframe.
+
+    Args:
+        kickac_df: Dataframe containing the data
+        action_plane: Plane of the action
+        tune_plane: Plane of the tune
+
+
+    Returns:
+        Dictionary containing x,y, x_err and y_err
+
+    """
+    col_natq, col_natq_std = _get_ampdet_columns(corrected)
+
+    columns = {"x": COL_ACTION(action_plane),
+               "xerr": COL_ACTION_ERR(action_plane),
+               "y": col_natq(tune_plane),
+               "yerr": col_natq_std(tune_plane),
+               }
+
+    not_found = [cv for cv in columns.values() if cv not in kickac_df.columns]
+    if any(not_found):
+        raise KeyError(f"The following columns were not found in kick-file: '{str(not_found)}'")
+
+    data = kickac_df.loc[:, list(columns.values())]
+    data.columns = columns.keys()
+
+    if data.isna().any().any():
+        LOG.warn(
+            f"Amplitude Detuning data for Q{tune_plane} and J{action_plane} contains NaNs"
+        )
+        data = data.dropna(axis=0)
+    return data.to_dict('series')
+
+
+# Timed DataFrames -------------------------------------------------------------
+
+
+def get_timestamp_index(index):
+    return pd.Index([i.timestamp() for i in index])
+
+
+def read_timed_dataframe(path):
+    df = tfs.read(path, index=COL_TIME())
+    df.index = pd.Index([CERNDatetime.from_cern_utc_string(i) for i in df.index], dtype=object)
+    return df
+
+
+def write_timed_dataframe(path, df):
+    df = df.copy()
+    df.index = pd.Index([i.strftime(get_cern_time_format()) for i in df.index], dtype=str)
+    tfs.write(str(path), df, save_index=COL_TIME())
+
+
+def read_two_kick_files_from_folder(folder):
+    return merge_two_plane_kick_dfs(
+        *[read_timed_dataframe(os.path.join(folder, f'{KICK_NAME}{p.lower()}.tfs')) for p in PLANES]
+    )
+
+
+def merge_two_plane_kick_dfs(df_x, df_y):
+    df_xy = tfs.TfsDataFrame(pd.merge(df_x, df_y, how='inner', left_index=True, right_index=True))
+    if len(df_xy.index) != len(df_x.index) or len(df_xy.index) != len(df_y.index):
```

Comment:
another (maybe more precise?) check could be based on the headers (ie Date, cwd, etc.) to make sure they come from the same analysis

Reply:
As the tfs files are loaded with timestamps as index, this should be fine. I am more checking that all data is available then checking if this makes sense (maybe someone uses files with modified headers, or no headers at all..). Identical timestamps is the easiest but ultimate check anyway